core/domain/classifier_services.py

```diff
@@ -347,6 +348,75 @@ def mark_training_job_complete(job_id):
                                            feconf.TRAINING_JOB_STATUS_COMPLETE)
 
 
+def mark_training_job_failed(job_id):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def update_failed_jobs(training_jobs):
+    """Updates status of job_models to failed.
+
+    Args:
+        training_jobs: list. List of jobs to be updated as failed.
+    """
+    for training_job in training_jobs:
+        mark_training_job_failed(training_job.job_id)
+
+
+def fetch_next_job():
+    """Gets next job model in the job queue.
+
+    Returns:
+        ClassifierTrainingJobModel. Domain object of the next training Job.
+    """
+    classifier_training_jobs = []
+    cursor = None
+    valid_jobs = []
+    failed_jobs = []
+
+    while len(valid_jobs) == 0:
```

Comment:
OK, I have some questions about this (perhaps @prasanna08 could weigh in, too):
- What are you planning to do if a pending job doesn't return by TTL? Email the admin, discard the job, or retry the job?
  - If email the admin, what should the admin do in that case?
  - If discard the job, then does that mean there'll never be a classifier for that exp_id/version/state?
  - If retry the job, then at what point do we stop retrying?
- What is our strategy for retrieving jobs? Is it FIFO or LIFO or something else? Currently it doesn't really look like there's a strategy, but we should be opinionated about this.
  - Also, if the strategy is FIFO then consider the approach of having a last_checked_time field on the job model that's indexed. When a job is retrieved and sent to the VM, increment that last_checked_time field by TTL. When you do the query, order the results by the last_checked_time field. That way you process jobs in the order you want to process them.